Bottom: 2d7d36989522b9e010497256793038fb2cb5aa9e
Top:    7af76078e335d5acccea9e83cb4e86eea842ce74
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-26 15:35:32 -0700

rsockets: Fix shutdown

mpi finalize hangs - fix it

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/preload.c b/src/preload.c
index 52eaf1a..515cc10 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -798,9 +798,13 @@ int select(int nfds, fd_set *readfds, fd_set *writefds,
 	return ret;
 }
 
+#include <stdio.h>
+
 int shutdown(int socket, int how)
 {
 	int fd;
+	printf("shutdown how %s\n", how == SHUT_RDWR ? "RDWR" : how == SHUT_RD ? "RD" : "WR");
+	fflush(stdout);
 	return (fd_get(socket, &fd) == fd_rsocket) ?
 		rshutdown(fd, how) : real.shutdown(fd, how);
 }
